### PR TITLE
Simplify replication worker retry logic

### PR DESCRIPTION
**Pull request by samarabbas**
Current retry logic can result in some replication tasks to be retried
indifinitely if we have bugs in our history replication code.
Updated the logic to make it much simpler.  It now uses a retryable
history client to retry any history service transient errors before
returning the error back to processWithRetry pump.  The pump will keep
on retrying whiltelisted transient errors for ever as it makes no sense
to move those messages to DLQ, otherwise it decrements the remaining
retry count and continue applying the message until fixed value of max
retry count is hit for that task.

### CODE REVIEW

File: service/worker/processor.go [hunk 1]
```diff
@@ -267,54 +196,55 @@ func (p *replicationTaskProcessor) messageProcessLoop(workerWG *sync.WaitGroup,
...
+	remainingRetryCount := p.config.ReplicationTaskMaxRetry
 
 ProcessRetryLoop:
-	for {
+	for attempt := 0; ; attempt++ {
 		select {
 		case <-p.shutdownCh:
 			return
 		default:
 			op := func() error {
```
Comment (wxing1292) [inline service/worker/processor.go]: op should be move outside of the for loop

File: service/worker/processor.go [hunk 2]
```diff
@@ -456,6 +393,14 @@ func (p *replicationTaskProcessor) updateFailureMetric(scope int, err error) {
 	}
 }
 
+func (p *replicationTaskProcessor) isRetryTaskError(err error) bool {
+	if _, ok := err.(*shared.RetryTaskError); ok {
```
Comment (wxing1292) [inline service/worker/processor.go]: maybe this should also include the timeout error

Comment (samarabbas) [reply service/worker/processor.go]: Timeout is already covered by transient retry policy.

File: service/worker/processor.go (hunk 1 above)
Comment (samarabbas) [reply service/worker/processor.go]: Good point.  Moved outside.

**Pull request merged**